src/main/java/com/internet/shop/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,222 @@
+            statement.setString(1, car.getModel());
+            statement.setLong(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t insert this car:" + car + "into DB ", e);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getQuery = "SELECT car_id, car_model, m.manufacture_id, "
+                + "m.manufacture_name, m.manufacture_country "
+                + "FROM cars c "
+                + "JOIN manufacturers m ON c.manufacturer_id = m.manufacture_id "
+                + "WHERE c.car_id = ? AND c.deleted = FALSE;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(getQuery)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = createCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get data with id:"
+                    + id + " from DB ", e);
+        }
+        if (car != null) {
+            car.setDrivers(getDrivers(car.getId()));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateQuery = "UPDATE cars SET manufacturer_id = ?, car_model = ? "
+                + "WHERE car_id = ? AND deleted = false";
+        String deleteQuery = "DELETE FROM drivers_cars WHERE car_id = ?";
+        String insertQuery = "INSERT INTO drivers_cars(driver_id, car_id) VALUES (?, ?)";
+        Long carId = car.getId();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCars = connection.prepareStatement(updateQuery);
+                PreparedStatement deleteDrivers =
+                        connection.prepareStatement(deleteQuery);
+                PreparedStatement insertDrivers =
+                        connection.prepareStatement(insertQuery)) {
+            updateCars.setLong(1, car.getManufacturer().getId());
+            updateCars.setString(2, car.getModel());
+            updateCars.setLong(3, carId);
+            updateCars.executeUpdate();
+            updateCars.close();
+            deleteDrivers.setLong(1, car.getId());
+            deleteDrivers.executeUpdate();
+            deleteDrivers.close();
+            for (Driver driver : car.getDrivers()) {
+                insertDrivers.setLong(1, driver.getId());
+                insertDrivers.setLong(2, car.getId());
+                insertDrivers.executeUpdate();
+                insertDrivers.close();
```

Comment:
if we close the statement in a loop, will we be able to proceed?

I would still recommend to move the logic of removing from cars_drivers and inserting to cars_drivers into 2 separate methods